app/code/core/Mage/Checkout/controllers/CartController.php

```diff
@@ -711,4 +714,15 @@ protected function _getCustomerSession()
     {
         return Mage::getSingleton('customer/session');
     }
+
+    /**
+     * Set product form data in checkout session for populating the product form
+     * in case of errors in add to cart process.
```

Comment:
```suggestion
     * in case of errors in add to cart process.
     * @return void
```

dunno if it's a good thing to add the return in the doc